src/main/scala/org/broadinstitute/dsde/workbench/sam/service/PolicyEvaluatorService.scala

```diff
@@ -92,6 +95,18 @@ class PolicyEvaluatorService(
     } yield res
   }
 
+  private def isMemberOfAllAuthDomainGroups(authDomains: NonEmptyList[WorkbenchGroupName], userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[Boolean] = {
```

Comment:
Could this be accomplished by using `directoryDAO.listIntersectionGroupUsers` and checking to see if the user is in it? I'm not sure how that would compare from an efficiency perspective, but it seems like it'd be a more straightforward way to find out if the user is in all the auth domain groups

Reply:
I think `listIntersectionGroupUsers` will fail if there is only 1 group (it is meant to intersect the policy group with all the AD groups, not the AD groups alone). However it _might_ scale better when there are 2 or more AD groups. Also my initial implementation which just replaced the body of this function with
```
directoryDAO.listIntersectionGroupUsers(authDomains.toList.toSet, samRequestContext).map(_.contains(userId))
```
failed some unit tests so I am going to save this suggestion for future consideration.